server/apps/microservices/src/processors/video-transcode.processor.ts

```diff
@@ -71,7 +72,7 @@ export class VideoTranscodeProcessor {
   async runVideoEncode(asset: AssetEntity, savedEncodedPath: string): Promise<void> {
     const config = await this.systemConfigService.getConfig();
 
-    if (config.ffmpeg.transcodeAll) {
+    if (config.ffmpeg.transcode == TranscodePreset.ALL) {
```

Comment:
Would you mind changing the flow of the method to be more of something like

let transcode: boolean

if(all) 
  transcode = true

Else if ...
  run Ffprobe pipeline
  transcode = ...

Else
  transcode = false
  ...

And finally all conditions run this code:

If transcode
  Run pipeline

I think keeping the method linear with less branching and return statements makes it more readable. It might be even better to extract that logic into a method `needsTranscoding(asset): boolean`

Reply:
Sure. I will refactor this method

Reply:
Awesome, thanks.

Reply:
Done. Please review